data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/auth/OIDCDiscoveryConfigListener.java

```diff
@@ -62,14 +62,10 @@ public OIDCDiscoveryConfigListener(String featuresConfigPath, Vertx vertx, int t
     }
 
     public OIDCDiscoveryConfig getOidcDiscoveryConfig() {
-        return oidcDiscoveryConfig;
+        return oidcDiscoveryConfig.get();
     }
 
     public int registerCallback(Consumer<OIDCDiscoveryConfig> callback) {
```

Comment:
```suggestion
    public synchronized int registerCallback(Consumer<OIDCDiscoveryConfig> callback) {
```

since executing the 2 `add + size() - 1` can lead to return the wrong index if one thread is executing `add` before `size() - 1` is executed